tests/jlm/tooling/TestJlmOptCommandLineParser.cpp

```diff
@@ -164,6 +139,11 @@ OutputFormatParsing()
   for (size_t n = start; n != end; n++)
   {
     auto outputFormat = static_cast<JlmOptCommandLineOptions::OutputFormat>(n);
+#ifndef ENABLE_MLIR
```

Comment:
The PR title is about this fix, but the majority of the changes are related to the cstrings.
It might be better to split it into two PRs, as they are not related.

Reply:
Yes, I reverted those changes to put them in a separate PR